src/Service/DatabaseFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Meanbee\Magedbm2\Service;
+
+use DI\Container;
+
+class DatabaseFactory
+{
+    /**
+     * @param Container $container
+     * @return DatabaseInterface
+     * @throws \Psr\Container\ContainerExceptionInterface
+     * @throws \Psr\Container\NotFoundExceptionInterface
+     */
+    public static function create(Container $container)
+    {
+        switch ($container->get('database_adapter')) {
+            case 'shell':
+            default:
+                $instance = $container->make(\Meanbee\Magedbm2\Service\Database\Shell::class);
```

Comment:
It would be nice to eventually move all of these to `di.php` directly.

Reply:
When I first wrote this I did have everything in `di.php` but decided to move it out to factories! I thought it was a bit cleaner, though I'm not a fan of passing the container around.